### PR TITLE
TopAppBar: implement support for keeping scroll behavior state

### PR DESCRIPTION
**Pull request by hrach**

### CODE REVIEW

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt [hunk 1]
```diff
@@ -66,24 +72,69 @@ public interface TopAppBarScrollBehavior {
...
 }
 
+private class TopAppBarScrollBehaviorSaver(
+    private val factory: () -> TopAppBarScrollBehavior,
+) : Saver<TopAppBarScrollBehavior, List<Float>> {
+    override fun restore(value: List<Float>): TopAppBarScrollBehavior =
+        factory().apply {
+            contentOffset = value[0]
+            offsetLimit = value[1]
+            offset = value[2]
```
Comment (gerak-cz) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:127]: nit: Maybe order these as they are defined in the interface?

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt [hunk 2]
```diff
@@ -66,24 +72,69 @@ public interface TopAppBarScrollBehavior {
...
+        }
+
+    override fun SaverScope.save(value: TopAppBarScrollBehavior): List<Float> =
```
Comment (gerak-cz) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:130]: nit: The order of the overridden methods is swapped.

**Review by gerak-cz — APPROVED** [review]

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt [hunk 3]
```diff
@@ -66,24 +72,69 @@ public interface TopAppBarScrollBehavior {
     public var contentOffset: Float
 
     public companion object {
-        public fun pinnedScrollBehavior(
+        @Composable
+        public fun rememberPinned(
```
Comment (gerak-cz) [inline ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:76]: Do we want this nested in the type? Isn't the convention of remember functions to be top level?

`rememberPinnedScrollBehavior()` ?

Comment (hrach) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:76]: I do not know about this convention. It seems quite ok to have it united under a single "namespace".

Comment (hrach) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:76]: I do not know about this convention. It seems quite ok to have it united under a single "namespace".

Comment (gerak-cz) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt:76]: Well it was my question. So far all the special remember functions I saw were top level.

If that is not the convention then this is probably fine.

File: ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt (hunk 1 above)
Comment (hrach) [reply ui/src/main/java/kiwi/orbit/compose/ui/controls/TopAppBarScrollBehavior.kt]: fixed

**Pull request merged**